tools/oas-configurable-proxy/index.js

```diff
@@ -0,0 +1,146 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+const SwaggerParser = require('@apidevtools/swagger-parser');
+const minimist = require('minimist')
+const fs = require('fs')
+const path = require('path');
+const handlebars = require('handlebars');
+
+const argv = minimist(process.argv.slice(2));
+const outFolder = argv['out']
+const oasPath = argv['oas']
+const basepath = argv['basepath']
+const name = argv['name']
+const envs = argv['envs']
+
+generateFromOAS({ outFolder, oasPath, basepath, name, envs });
+
+/**
+ * Generate Proxy Config from OAS
+ *
+ * @param {Object} config OAS generator config
+ */
+async function generateFromOAS(config) {
+    try {
+        const api = await SwaggerParser.validate(config.oasPath);
+        api.openapi
+        const oasVersionString = api.openapi || api.swagger;
+        const oasVersionSplit = oasVersionString.split(".");
+        const oasVersionMajor = parseInt(oasVersionSplit[0], 10);
+        const proxyName = config.name || api.info.title.replace(/ /g, '-').toLowerCase();
+        const basepath = config.basepath || `/${proxyName}/v1`
+
+        // target service
+        let targetService = 'https://notfound.apigee.google.com # TODO replace me'
```

Comment:
Is the # intentionally inside the string or is that supposed to be a JS line comment? :)

Reply:
it's a comment to go into the templated YAML ;)